### PR TITLE
Custom NFT contract

### PR DESCRIPTION
**Pull request by aditya-passbird**

### CODE REVIEW

File: lambda/voting/scripts/create.js [hunk 1]
```diff
@@ -1,8 +1,7 @@
 // Script to deploy Smart contract using Metakeep Lambda and REST API.
-import fetch from "node-fetch";
 import fs from "fs";
 import env from "dotenv";
-import { exit } from "process";
+import { create } from "../../lambdaUtils.mjs";
```
Comment (rohan-passbird) [inline lambda/voting/scripts/create.js:4]: move voting changes to a separate PR. I need more time to review the NFT PR

File: lambda/custom-nft/contracts/contracts/MetaKeepCertificate.sol [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        uint256 certificateId; // this is the id of the certificate ie token Id
+    }
+
+    address public owner;
+    uint256 public certificateCount = 0;
+
+    mapping(uint256 => CertificateHolder) public certificateHolders;
+    // This is the constructor whose code is
+    // run only when the contract is created.
+    constructor(string memory symbol, address lambdaOwner, string memory lambdaName) ERC721(lambdaName, symbol) MetaKeepLambda(lambdaOwner, lambdaName) {
```
Comment (rohan-passbird) [inline lambda/custom-nft/contracts/contracts/MetaKeepCertificate.sol:28]: formatting looks wrong. Remove sol files from prettier ignore

File: lambda/custom-nft/contracts/contracts/MetaKeepCertificate.sol [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+pragma solidity ^0.8.9;
+
+// We import this library to be able to use console.log
+import "hardhat/console.sol";
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "metakeep-lambda/ethereum/contracts/MetaKeepLambda.sol";
+
+
+// This is the main building block for smart contracts.
+contract MetaKeepCertificates is ERC721, MetaKeepLambda {
```
Comment (rohan-passbird) [inline lambda/custom-nft/contracts/contracts/MetaKeepCertificate.sol:14]: file name and contract name do not match

File: lambda/custom-nft/contracts/contracts/MetaKeepCertificate.sol [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+    struct CertificateHolder {
+        string name;
+        string email;
+        uint256 certificateId; // this is the id of the certificate ie token Id
+    }
+
+    address public owner;
```
Comment (rohan-passbird) [inline lambda/custom-nft/contracts/contracts/MetaKeepCertificate.sol:22]: where is this being used

File: lambda/custom-nft/scripts/.env [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+// Your lambda contract address.
+// You get this after running the create script.
+LAMBDA_ADDRESS = 
+
+// The email of the candidate who will be issued certificate
+// You can also use an eth address here 
+// after modifying the mintAndDestroyCertificate.mjs script.
+CANDIDATE_EMAIL = 
+
```
Comment (rohan-passbird) [inline lambda/custom-nft/scripts/.env:12]: wrong

File: lambda/custom-nft/scripts/mintAndDestroyCertificate.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  // Checks if the API_KEY is set in the .env file.
+  checkAPIKey();
+
+  // Gets the user Address to whom you want to issue certificate.
+  const userAddress = await getUserWallet(process.env.CANDIDATE_EMAIL);
+  const tokenId = Math.floor(Math.random() * 10000);
+
+  /* *************************************************************** Issue Certificate *************************************************************** */
+  console.log(
+    "***************************************************** Issue Certificate *****************************************************\n"
```
Comment (rohan-passbird) [inline lambda/custom-nft/scripts/mintAndDestroyCertificate.js:22]: why are these lines so long. Lines should not be more than 100 chars long

File: lambda/custom-nft/scripts/.env [hunk 7]
```diff
@@ -0,0 +1,12 @@
+// Your MetaKeep API key
+API_KEY = 
+
+// Your lambda contract address.
+// You get this after running the create script.
+LAMBDA_ADDRESS = 
+
+// The email of the candidate who will be issued certificate
+// You can also use an eth address here 
+// after modifying the mintAndDestroyCertificate.mjs script.
```
Comment (rohan-passbird) [inline lambda/custom-nft/scripts/.env:10]: comment wrong

File: lambda/custom-nft/contracts/contracts/MetaKeepNFT.sol [hunk 8]
```diff
@@ -0,0 +1,110 @@
+//SPDX-License-Identifier: UNLICENSED
+
+// Solidity files have to start with this pragma.
+// It will be used by the Solidity compiler to validate its version.
+pragma solidity ^0.8.9;
+
+// We import this library to be able to use console.log
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
```
Comment (aditya-passbird) [inline lambda/custom-nft/contracts/contracts/MetaKeepNFT.sol:9]: @rohan-passbird here we are inherting the OZ's Interface to make calls from this contract. Ideally it should be metakeep's ERC20 Interface.
